### ISSUE TITLE
process killed during training after step 100

### ISSUE THREAD

**Issue by ProudHoosier** [dennybritz/cnn-text-classification-tf#104]
Evaluation:
Killed
After 100 steps(), the program is killed automatically, while evaluating against validation data.
tf.flags.DEFINE_integer("evaluate_every", 100, "Evaluate model on dev set after this many steps (default: 100)")
Please help.

**Comment by Alix-Martin** [comment]
this is because the Evaluation on the dev samples is done in one batch, that in your case does not fit in memory. 

The quick fix is to lower the number of dev samples using the --dev_sample_percentage option. For instance --dev_sample_percentage 0.05 uses 5% of samples instead of the default 10%.

A better way would be to split the dev samples into batches of the same size as on the train samples, and to aggregate the statistics.

**Comment by zcleghern** [comment]
Why would this be enough memory usage to cause the program to crash? I get the same error on my laptop (with plenty of memory) but bumping the dev set down to 5% makes it work. I'd like to solve the problem outright instead of using a workaround, since I'm adapting this code to use 10-fold cross validation (and if you want me to submit a pull request to merge that in I'd be happy to) so I'd really like to use 10% for the validation set.

**Comment by nazhang8** [comment]
@Alix-Martin, in order to split the dev samples, I changed the code from 

dev_step(x_dev, y_dev, writer=dev_summary_writer)

to 

dev_step(x_batch, y_batch, writer=dev_summary_writer)

Is it the right way to do? And how to aggregate the evaluation accuracy? 

(My dataset is very large, even I use split ratio 1%, dev set still couldn't be fitted into memory)

**Comment by nazhang8** [comment]
@Alix-Martin 

How to split the dev samples into batches? Could you give some hints? thanks.

**Comment by zcleghern** [comment]
`devBatches = data_helpers.batch_iter(
                    list(zip(dev_x, dev_y)), FLAGS.batch_size, 1)`

should give you the batch iterator you need. then you can just get each batch as normal

**Comment by scorpjio** [comment]
I just put some code in [this comment](https://github.com/dennybritz/cnn-text-classification-tf/issues/132#issuecomment-372669233) how I did the dev/test in batches
